vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher_list.proto

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+syntax = "proto3";
+
+option optimize_for = LITE_RUNTIME;
+
+package publishers_pb;
+
+message PublisherList {
+  enum CompressionType {
+    NO_COMPRESSION = 0;
+  }
+
+  // The size, in bytes, of each hash prefix stored in the
+  // prefix list. This value must be between 4 and 32.
+  uint32 prefix_size = 1;
+
+  // The type of compression applied to the hash prefix list.
+  CompressionType compression_type = 2;
```

Comment:
This approach of compressing the data inside of the protobuf message is inspired by Safe Browsing (see [here](https://cs.chromium.org/chromium/src/components/safe_browsing/core/db/safebrowsing.proto?q=safebrowsing.proto&sq=package:chromium&g=0&l=422)), which embeds Rice-encoded prefix delta values into the message when the prefix lengths are 4 bytes. This message format will give us the option of moving in that direction in the future if we choose.